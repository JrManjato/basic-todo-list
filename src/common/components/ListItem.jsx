import React from 'react'

export default function ListItem({ id, text, isCheckable, setList, list }) {
    
    // Deletes an item based on the `item.id` key
    function deleteItem(id) {
        const newArray = list.filter((item) => item.id !== id);
        setList(newArray);
    }

    function updateScale(id) {
        // Get the current item
        const currentItem = list.filter((item) => item.id === id);
    
        // Create a new item with the updated scale
        const newItem = {
          id: currentItem[0].id,
          value: currentItem[0].value,
          scale: 'done'
        };
    
        deleteItem(id);
    
        // Update the list
        setList((oldList) => [...oldList, newItem]);
      }

    return (
        <li key={id}>
            {isCheckable &&
                <input
                    type="checkbox"
                    id={text}
                    onChange={() => updateScale(id)}
                />
            }
            <label htmlFor={text}>{text}</label>
        </li>
    )
}
