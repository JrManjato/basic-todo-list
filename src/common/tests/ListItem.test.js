import { create } from "react-test-renderer";
import ListItem from "../components/ListItem";

describe("ListItem Test", () => {

    it("should renders the component", () => {
        const renderer = create(<ListItem />);

        expect(renderer).toBeTruthy();
    });

    test("should renders the component [snapshot]", () => {
        const listItem = create(<ListItem text isCheckable />);

        const json = listItem.toJSON();

        expect(json).toMatchSnapshot();
    });

});