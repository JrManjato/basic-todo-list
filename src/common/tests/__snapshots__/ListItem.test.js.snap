// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem Test should renders the component [snapshot] 1`] = `
<li>
  <input
    data-testid="checkbox"
    id={true}
    onChange={[Function]}
    type="checkbox"
  />
  <label
    htmlFor={true}
  />
</li>
`;
